### PR TITLE
SEP-12: Check customer fields status

### PR DESCRIPTION
**Pull request by quietbits**

### CODE REVIEW

File: src/methods/sep12/collectSep12Fields.ts [hunk 1]
```diff
@@ -43,12 +44,42 @@ export const collectSep12Fields = async ({
...
 
+  const fieldsToCollect = Object.entries(resultJson.fields ?? {}).reduce(
+    (collectResult: any, field: any) => {
+      const [key, props] = field;
+
+      if (
+        !props.status ||
+        [
+          Sep12CustomerFieldStatus.NOT_PROVIDED,
+          Sep12CustomerFieldStatus.REJECTED,
```
Comment (quietbits) [inline src/methods/sep12/collectSep12Fields.ts:55]: Do we want to include rejected fields as well?

Comment (JakeUrban) [reply src/methods/sep12/collectSep12Fields.ts:55]: If the customer's status is `REJECTED`, then that is final (the user will never pass). If the customer's status is `NEEDS_INFO`, then individual fields that are `REJECTED` can be re-collected.

It doesn't look like we're checking the customer status other than when the customer is new, so we'll need to adjust the logic here.

**Review by JakeUrban — CHANGES_REQUESTED** [review]
Looks good other than the topic of REJECTED fields you asked about.

Also, this is unrelated by I see on line 24 that we may be sending an `undefined` memo. If `memo` is not defined, we shouldn't be sending it in the request.

**Comment by quietbits** [comment]
@JakeUrban Thanks for your comments! The memo looks good, we're not submitting it if it's not provided (spreading `{}` won't produce anything in this case). It was good to double check. 👍

File: src/methods/sep12/collectSep12Fields.ts [hunk 2]
```diff
@@ -37,18 +38,47 @@ export const collectSep12Fields = async ({
...
   if (resultJson.fields) {
     log.instruction({
-      title: "Received the following fields",
+      title: "Received the following customer fields",
       body: resultJson.fields,
     });
   }
 
-  return resultJson.fields;
+  if (Object.keys(fieldsToCollect).length > 0) {
```
Comment (piyalbasu) [inline src/methods/sep12/collectSep12Fields.ts:72]: sooooo small - but this could be simplified to `(Object.keys(fieldsToCollect).length )`

**Review by piyalbasu — APPROVED** [review]

**Review by JakeUrban — APPROVED** [review]
We will need to make some adjustments to this logic in the near future once we make changes to the SEP-12 `GET /customer` response format but this looks good for what SEP-12 is now 👍

**Pull request merged**